client/pages/background.jsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { parseRoute } from '../lib';
+
+export default class Background extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      route: parseRoute(window.location.hash)
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener('hashchange', () => {
+      this.setState({ route: parseRoute(window.location.hash) });
+    });
+  }
```

Comment:
You don't want to have a separate event listener and state here for your routes. If this component needs the route information it should be passed from the Home component via props.